borbskeeper/Features/Profile/ProfileViewController.m

```diff
@@ -0,0 +1,162 @@
+//
+//  ProfileViewController.m
+//  borbskeeper
+//
+//  Created by rodrigoandrade on 7/25/19.
+//  Copyright © 2019 juliapark628. All rights reserved.
+//
+
+#import "ProfileViewController.h"
+#import "UIImageView+AFNetworking.h"
+#import "BorbParseManager.h"
+#import "Task.h"
+#import "CompletedTasksListInfiniteScrollView.h"
+#import "TaskCell.h"
+
+@interface ProfileViewController ()<UIImagePickerControllerDelegate, UINavigationControllerDelegate, InfiniteScrollDelegate>
+
+@property (weak, nonatomic) IBOutlet UIImageView *profilePicture;
+@property (weak, nonatomic) IBOutlet UILabel *userName;
+@property (weak, nonatomic) IBOutlet CompletedTasksListInfiniteScrollView *completedTasksListInfiniteScrollView;
+@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activityIndicator;
+
+@property (strong, nonatomic) NSMutableArray *completeTaskList;
+@property (strong, nonatomic) NSDate *latestDate;
+@property (strong, nonatomic) UIImage *originalImage;
+@property (strong, nonatomic) UIImage *editedImage;
+@property (nonatomic, strong) UIRefreshControl *refreshControl;
+
+@end
+
+@implementation ProfileViewController
+
+static NSString *const USER_PROF_PIC_KEY = @"profilePicture";
+static NSString *const COMPLETE_TASK_TABLE_VIEW_CELL_ID = @"CompletedTaskCell";
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.completedTasksListInfiniteScrollView.infiniteScrollDelegate = self;
+    [self setupProfile];
+    [self fetchData];
+    [self refreshCompleteTaskList];
```

Comment:
we should start animating the activity indicators before fetching the data, let's say fetchData is fast and it finishes right before we call startAnimating, the activity indicator will keep animating until next reload.

Reply:
This has been fixed